mobile-browser-based-version/mobile-app/src/components/TitanicComponents/titanic_training.vue

```diff
@@ -0,0 +1,1018 @@
+                  +d[headerCopied[0]], // PassengerId
+                  +d[headerCopied[1]], // Survived
+                  d[headerCopied[3]] == "male" ? 1 : 0, // Sex
+                  +d[headerCopied[4]], // Age
+                  +d[headerCopied[5]], // SibSp
+                  +d[headerCopied[6]], // Parch
+                  +d[headerCopied[8]], // Fare
+                  +d[headerCopied[11]], // Pclass
+                ];
+              },
+              function (error, rows) {
+                console.log(rows);
+              }
+            );
+
+            let ycsv = d3.csvParse(
+              e.target.result,
+              function (d) {
+                return +d[headerCopied[1]];
+              },
+              function (error, rows) {
+                console.log(rows);
+              }
+            );
+            /*
+                return {
+                    // Map the user's data point to our format 
+                    // and convert what can be converted to int or string (with +) 
+                    PassengerId: +d[headerCopied[0]],
+                    Survived: +d[headerCopied[1]],
+                    Name: d[headerCopied[2]],
+                    Sex: d[headerCopied[3]],
+                    Age: +d[headerCopied[4]],
+                    SibSp: +d[headerCopied[5]],
+                    Parch: +d[headerCopied[6]],
+                    Ticket: d[headerCopied[7]],
+                    Fare: +d[headerCopied[8]],
+                    Cabin: d[headerCopied[9]],
+                    Embarked: d[headerCopied[10]],
+                    Pclass: +d[headerCopied[11]]
+                };*/
+
+            // console.log(tf.tensor2d(Xcsv));
+            // console.log(tf.tensor1d(ycsv));
+
+            let Xtrain = tf.tensor2d(Xcsv);
+            let ytrain = tf.tensor1d(ycsv);
+
+            //let df = new DataFrame(parsedCSV)
+            // console.log(df)
+            this.train(Xtrain, ytrain);
+          } else {
+            console.log("Cannot Start training");
+          }
+        };
+
+        reader.readAsText(file);
+      }
+    },
+
+    /**
+     * Creates the tf.model
+     * TO DO: fetch model from local storage
+     * @returns Returns a tf.model for training
+     */
+    get_model() {
+      const model = tf.sequential();
+      model.add(
+        tf.layers.dense({
+          inputShape: [8],
+          units: 124,
+          activation: "relu",
+          kernelInitializer: "leCunNormal",
+        })
+      );
+      model.add(tf.layers.dense({ units: 64, activation: "relu" }));
+      model.add(tf.layers.dense({ units: 32, activation: "relu" }));
+      model.add(tf.layers.dense({ units: 1, activation: "sigmoid" }));
+      model.summary();
+      return model;
+    },
+
+    /**
+     * Function called once data has been loaded and processed. Updates the model 
+     * stored in local memory
+     
+     * @returns Returns a tf.model for training
+     */
+    async train(Xtrain, ytrain) {
```

Comment:
towards separating UI & training algorithm: could you externalize this function train() function into its own .js file? it would need to get the model and UI pointer as well somehow, so that the callback can update the accuracy display? later we could then possibly give a flag to the trainer what kind of training we want (distributed or local training), but we can make a new PR for that later.